Made SimpleJson optional

baseplate-admin opened the pull request:
Closes #181

baseplate-admin committed:
Modernize

baseplate-admin committed:
Update setup.py

Made SimpleJson optional

baseplate-admin committed:
Update README.rst

asfaltboy reviewed (changes requested):
My apologies for slow response. Thanks for this PR, I do think it makes sense.

I think we can take it a step further and just remove simplejson entirely now. I say this because it seems we only used simplejson to support python 2.7/3.3, as per [their docs](https://simplejson.readthedocs.io/en/latest/):
> [simplejson](https://simplejson.readthedocs.io/en/latest/#module-simplejson) exposes an API familiar to users of the standard library marshal and pickle modules. It is the externally maintained version of the json library, but maintains compatibility with the latest Python 3.8+ releases back to Python 3.3 as well as the legacy Python 2.5 - Python 2.7 releases.

Before we apply merge the change, though, could we add a test case or two to  `advanced_filters/tests/test_q_serializer.py` which ensure serialising and unserialising date and datetime values work?

baseplate-admin commented:
Hi, Thanks for your response. I am a bit busy now. Will swing back to this PR soon.

---

> I think we can take it a step further and just remove simplejson entirely now. I say this because it seems we only used simplejson to support python 2.7/3.3, as per [their docs](https://simplejson.readthedocs.io/en/latest/):

I also think `stdlib`'s json does make more sense to support.

> Before we apply merge the change, though, could we add a test case or two to advanced_filters/tests/test_q_serializer.py which ensure serialising and unserialising date and datetime values work?

Sure gimme some time

baseplate-admin committed:
Merge remote-tracking branch 'upstream/develop' into develop

baseplate-admin committed:
Update test_q_serializer.py

Fix : https://github.com/modlinltd/django-advanced-filters/pull/182#pullrequestreview-1425894485

baseplate-admin commented:
Hi, sorry i forgot about this PR.

https://github.com/modlinltd/django-advanced-filters/pull/182#pullrequestreview-1425894485 has been addressed. Please take a look.

baseplate-admin committed:
Update setup.py

Update setup.py
